packages/neuron-wallet/src/services/ckb-runner.ts

```diff
@@ -142,18 +142,24 @@ export const startCkbNode = async () => {
     }
   })
 
-  ckb.on('error', error => {
+  currentProcess.on('error', error => {
     logger.error('CKB:\trun fail:', error)
     isLookingValidTarget = false
-    ckb = null
+    if (ckb?.pid === currentProcess.pid) {
```

Comment:
`Object.is` should be used here based on the assignment at line 161

```js
ckb = currentProcess
```

And the `ckb = null` when `ckb?.pid === currentProcess.pid` is a bit weird because it implies there will be 2 processes spawned by Neuron

What if send an `exit` signal to the previous one, then spwan a new one.

Besides, why not `ckb = currentProcess` when `ckb.pid !== currentProcess.pid`

Reply:
If the `ckb` is set as null by another event, the ckb can't stop forever by Neuron.
When the user switches the node faster, sometimes the `ckb.pid !== currentProcess.pid`.

Reply:
> If the `ckb` is set as null by another event, the ckb can't stop forever by Neuron. When the user switches the node faster, sometimes the `ckb.pid !== currentProcess.pid`.

If so, I think the fix should be replacing `set ckb as null` with `stop ckb`

Reply:
https://github.com/nervosnetwork/neuron/blob/develop/packages/neuron-wallet/src/services/ckb-runner.ts#L160-L171
The ckb has been set as null when stopping the ckb, and it will send a `close` event to the `currentProcess`, so here should add a judgment to avoid setting another ckb to null.